Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3390,19 +3390,43 @@ void WebPageProxy::dragExited(DragData& dragData)
     performDragControllerAction(DragControllerAction::Exited, dragData);
 }
 
-void WebPageProxy::performDragOperation(DragData& dragData, const String& dragStorageName, SandboxExtension::Handle&& sandboxExtensionHandle, Vector<SandboxExtension::Handle>&& sandboxExtensionsForUpload)
+#if PLATFORM(COCOA)
+void WebPageProxy::propagateDragAndDrop(FrameIdentifier frameID, RemoteUserInputEventData remoteUserInputEventData, CompletionHandler<void(bool)>&& completionHandler)
 {
+    RefPtr parent = WebFrameProxy::webFrame(frameID);
+    RefPtr targetFrame = WebFrameProxy::webFrame(remoteUserInputEventData.targetFrameID);
+    sendWithAsyncReplyToProcessContainingFrame(remoteUserInputEventData.targetFrameID, Messages::WebPage::PerformDragOperation(remoteUserInputEventData.targetFrameID, internals().currentDragOperationData->dragData), [completionHandler = WTFMove(completionHandler)](bool handled) mutable {
+        completionHandler(handled);
+    });
+}
+#endif
+
 #if PLATFORM(COCOA)
-    grantAccessToCurrentPasteboardData(dragStorageName);
+void WebPageProxy::fetchSandboxExtensionsForDragAction(FrameIdentifier frameID, CompletionHandler<void(SandboxExtension::Handle&&, Vector<SandboxExtension::Handle>&&)>&& completionHandler)
```

Comment:
I think this should at least verify that the process asking for sandbox extensions is hosting the frame sent in this message.